clang/lib/Parse/ParseExprCXX.cpp

```diff
@@ -2203,6 +2203,12 @@ Parser::ParseCXXCondition(StmtResult *InitStmt, SourceLocation Loc,
       return ParseCXXCondition(nullptr, Loc, CK, MissingOK);
     }
 
+    EnterExpressionEvaluationContext Eval(
```

Comment:
This doesn't quite seem right to me, as it is covering parsing at least 2 things, do we really intend that?

Reply:
Do you mean using the extra context for this case would introduce unnecessary overhead? Should the `if constexpr` be parsed within the same context, but with different parsing logic?

Reply:
No, more that this applies to the expressoin on 2213 as well as the CXXCondition on 2221 (which looks like a recursive call?).  I just want to make sure we're not applying this context to stuff that it shouldn't be here.

Reply:
This context will be applied only for `ConstexprIf`. It uses the `ShouldEnter` flag to control the pushing of the extra context, all other kinds will be ignored

https://github.com/llvm/llvm-project/blob/e1c1e74a6fd71dd889155100d4c0f5e3284f7a22/clang/include/clang/Sema/EnterExpressionEvaluationContext.h#L29-L33

Reply:
Right, I saw that/understand how that 'ShouldEnter' works.  its just not clear to me that this is the correct place that the context should be introduced.

It isn't clear to me why it is introduced only for `Expression` and not an `InitStmtDecl` for example.  And also why it should apply to the expression AND condition (on 2213 and 2221 respectively) but not the condition on 2203, etc.